json/src/main/java/com/basistech/rosette/dm/jackson/EntityMixin.java

```diff
@@ -0,0 +1,36 @@
+/*
+* Copyright 2014 Basis Technology Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.basistech.rosette.dm.jackson;
+
+import com.basistech.rosette.dm.Mention;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link com.basistech.rosette.dm.ResolvedEntity}
+ */
+public abstract class EntityMixin {
+    @JsonCreator
+    EntityMixin(@JsonProperty("entityId") String entityId,
+                @JsonProperty("mentions") List<Mention> mentions,
+                @JsonProperty("confidence") Double confidence,
+                @JsonProperty("extendedProperties") Map<String, Object> extendedProperties) {
+        //
```

Comment:
I'm okay with type at the Entity level; we just don't have it today.
But I think REX could produce mentions with the same string but
different type.  There are "consistency features" that try to avoid
this, but I thought they were optional.  On the other hand, I think
indoc will only chain within a type, so if REX did produce "Boston"
as LOCATION and PERSON, they would already be separate entities
if indoc is run.

If indoc is not run, then we'd have one Entity per Mention, and then the
mention's type comes from it's Entity.

So, I think this would work.  But I'd wait for Itai to comment before proceeding.